airflow/providers/amazon/aws/triggers/eks.py

```
@@ -0,0 +1,178 @@
+    :param aws_conn_id: The Airflow connection used for AWS credentials.
+    """
+
+    def __init__(
+        self,
+        cluster_name: str,
+        fargate_profile_name: str,
+        poll_interval: int,
+        max_attempts: int,
+        aws_conn_id: str,
+    ):
+        self.cluster_name = cluster_name
+        self.fargate_profile_name = fargate_profile_name
+        self.poll_interval = poll_interval
+        self.max_attempts = max_attempts
+        self.aws_conn_id = aws_conn_id
+
+    def serialize(self) -> tuple[str, dict[str, Any]]:
+        return (
+            self.__class__.__module__ + "." + self.__class__.__qualname__,
+            {
+                "cluster_name": self.cluster_name,
+                "fargate_profile_name": self.fargate_profile_name,
+                "poll_interval": str(self.poll_interval),
+                "max_attempts": str(self.max_attempts),
+                "aws_conn_id": self.aws_conn_id,
+            },
+        )
+
+    @cached_property
+    def hook(self) -> EksHook:
+        return EksHook(aws_conn_id=self.aws_conn_id)
+
+    async def run(self):
+        async with self.hook.async_conn as client:
+            attempt = 0
+            waiter = client.get_waiter("fargate_profile_active")
+            while attempt < int(self.max_attempts):
+                attempt += 1
+                try:
+                    await waiter.wait(
+                        clusterName=self.cluster_name,
+                        fargateProfileName=self.fargate_profile_name,
+                        WaiterConfig={"Delay": int(self.poll_interval), "MaxAttempts": 1},
+                    )
+                    break
+                except WaiterError as error:
+                    if "terminal failure" in str(error):
+                        yield TriggerEvent(
+                            {"status": "failure", "message": f"Create Fargate Profile failed: {error}"}
+                        )
```

you should raise directly here instead of returning the error to the operator
see https://apache-airflow.slack.com/archives/CCPRP7943/p1685663968275419